csharp_dotnetcore/3.Welcome-Users/WelcomeUserBot.cs

```diff
@@ -0,0 +1,146 @@
+
+        // the bot state accessor object. Use this to access specific state properties
+        private readonly WelcomeUserStateAccessors _welcomeUserStateAccessors;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WelcomeUserBot"/> class.
+        /// Initalize WelcomeUserBot.
+        /// </summary>
+        /// <param name="statePropertyAccessor"> Bot state accessor object.</param>
+        public WelcomeUserBot(WelcomeUserStateAccessors statePropertyAccessor)
+        {
+            _welcomeUserStateAccessors = statePropertyAccessor ?? throw new System.ArgumentNullException("state accessor can't be null");
+        }
+
+        /// <summary>
+        /// Every Conversation turn for our WelcomeUser Bot will call this method, including
+        /// any type of activiaties such as ConversationUpdate or ContactRelationUpdate which
+        /// are sent when a user joins a conversation.
+        /// This bot doesn't use any dialogs; it's "single turn" processing, meaning a single
+        /// request and response, with no stateful conversation between user input.
+        /// This bot does use UserState to keep track of first message users send to the bot.
+        /// </summary>
+        /// <param name="turnContext">A <see cref="ITurnContext"/> containing all the data needed
+        /// for processing this conversation turn. </param>
+        /// <param name="cancellationToken">(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = new CancellationToken())
+        {
+            // use state accessor to extract the didBotWelcomeUser flag
+            var didBotWelcomeUser = await _welcomeUserStateAccessors.DidBotWelcomedUser.GetAsync(turnContext, () => false);
+
+            if (turnContext.Activity.Type == ActivityTypes.Message && turnContext.Responded == false)
+            {
+                // Your bot should proactively send a welcome message to a personal chat the first time
+                // (and only the first time) a user initiates a personal chat with your bot.
+                if (didBotWelcomeUser == false)
+                {
+                    // update user state flag to reflect bot handled first user interaction.
+                    await _welcomeUserStateAccessors.DidBotWelcomedUser.SetAsync(turnContext, true);
+
+                    await turnContext.SendActivityAsync($"you first message ever to this bot ", cancellationToken: cancellationToken);
```

Comment:
Casing. Make this a valid sentence.

Reply:
In our samples, we've not been passing the cancellation token, as it adds too much noise to the samples. Please remove it from this call.